readme.txt
Defender Of Nothing, SpeedHack 2007
by David Capello

Website: http://www.davidcapello.com.ar/defnot/

This game is INCOMPLETE


 Tools and Libraries Used
-----------------------------------------------------------------------

 Compiler/IDE: Emacs, MinGW (GCC), make, Allegro Sprite Editor

 Platform: Windows

 Language: C++


 SpeedHack Rules
-----------------------------------------------------------------------

  * Evilness:
    You can send people to Hell (burning them), but it's not the
    objective (anyway you will get points for that).
  * Radius of Influence:
    The burning effect of the devil with the people.
  * Get a Grip:
    INCOMPLETE (...It should be: The friction of cars with the floor. And
    when you smash angels on the street.)
  * Ratio, Respect!
    The game support widescreen resolution and a square aspect one, to
    switch one to another press F11.
  * Reverse Stereotypes:
    Angels that want to kill people to send them to heaven, and a
    Devil that defend people. Angels with glasses. And a God that
    you wouldn't want to known (INCOMPLETE).
  * Dialogue:
    Thanks to the "Act of Generosity" rule, I threw away the Dialogue rule.

  Another important note: all files and graphics were made between the
  3 days period of competition. There are only one file that wasn't
  coded during the 3 days period, and it's:

    include/vector2d.hpp (I took it from an example of my Vaca library)

  (that linear algebra never changes :)


 About the Game
-----------------------------------------------------------------------

The Heaven is getting empty and boring, so God want more people and
send his angels to find them. You are the Devil, and must to defend
the people, anyway, extra points will give you if you send people to
Hell to keep them more safe.


 How to Play
-----------------------------------------------------------------------

 Keys
   Arrows    - move the Devil (fly and walk)
   Left CTRL - charges fire, and shoot when you release the key


 Credits
-----------------------------------------------------------------------

Programming and graphics by:

  David Capello


ChangeLog
2008-12-18  David A. Capello  <davidcapello@gmail.com>

	* Fixed a lot of bugs.

2007-09-17  David A. Capello  <dacap@users.sourceforge.net>  12:48 a.m.

        * Finished the angels abducting, levels, final, backgrounds, etc.

2007-09-16  David A. Capello  <dacap@users.sourceforge.net>  08:46 p.m.

	* Added menu and options to change widescreen/fullscreen.
	* src/sprite.cpp: Added to make pixel-collision for sprites.

2007-09-16  David A. Capello  <dacap@users.sourceforge.net>  02:06 a.m.

	* src/angel.cpp: Added the wings to the angel.

2007-09-16  David A. Capello  <dacap@users.sourceforge.net>  01:10 a.m.

	* src/scorer.cpp, src/angel.cpp: Added the graphics of the angels
	and the minimap in "scorer".

	* src/object.cpp: A little more of abstraction.

2007-09-15  David A. Capello  <dacap@users.sourceforge.net>  08:39 p.m.

	* src/person.cpp: The animation is done. Let's start with the
	scorer and graphics of angels.

2007-09-15  David A. Capello  <dacap@users.sourceforge.net>  07:39 p.m.

	* src/person.cpp: Added people.

2007-09-15  David A. Capello  <dacap@users.sourceforge.net>  04:27 p.m.

	* src/particle.cpp: Added particle system and fire to the player.

2007-09-15  David A. Capello  <dacap@users.sourceforge.net>  12:16 p.m.

	* Added tiles and better graphics (devil with two colors).
        * Blackout here in Santa Fe (some minutes only...).

2007-09-15  David A. Capello  <dacap@users.sourceforge.net>  01:59 a.m.

	* src/media.cpp, media/bedevil.dat, src/player.cpp: Drawn the main
	character, with movements to all sides and a couple of "fire"
	animations. Also drawn the tail movement. Created the
	datafile. Added the graphics to the player.

2007-09-14  David A. Capello  <dacap@users.sourceforge.net>  10:32 p.m.

	* src/gameplay.cpp: input, player, level, screenshot. The only code
	reutilized from other project (http://vaca.sourceforge.net/) is
	vector2d. All the "look-at" and scroll movement was made.

2007-09-14  David A. Capello  <dacap@users.sourceforge.net>  04:13 p.m.

	* src/main.cpp (game_loop): Done the main game loop. And the base
	classes: Game, GameState, Control. I'm very happy: I didn't make
	any copy & paste :)


include/ids.h
/* Allegro datafile object indexes, produced by dat v4.2.3 (SVN), MinGW32 */
/* Datafile: defnot.dat */
/* Date: Sun Sep 16 23:47:31 2007 */
/* Do not hand edit! */

#define ANGELS_PCX                       0        /* BMP  */
#define BG2_PCX                          1        /* BMP  */
#define BG3_PCX                          2        /* BMP  */
#define BG_PCX                           3        /* BMP  */
#define DEVIL_PCX                        4        /* BMP  */
#define DEVIL_WALKING00_PCX              5        /* BMP  */
#define DEVIL_WALKING01_PCX              6        /* BMP  */
#define DEVIL_WALKING02_PCX              7        /* BMP  */
#define DEVIL_WALKING03_PCX              8        /* BMP  */
#define DEVIL_WALKING04_PCX              9        /* BMP  */
#define DEVIL_WALKING05_PCX              10       /* BMP  */
#define DEVIL_WALKING06_PCX              11       /* BMP  */
#define DEVIL_WALKING07_PCX              12       /* BMP  */
#define DISQUAL_PCX                      13       /* BMP  */
#define FONT_PCX                         14       /* FONT */
#define PALETTE_PCX                      15       /* PAL  */
#define PEOPLE_PCX                       16       /* BMP  */
#define TAIL00_PCX                       17       /* BMP  */
#define TAIL01_PCX                       18       /* BMP  */
#define TAIL02_PCX                       19       /* BMP  */
#define TAIL03_PCX                       20       /* BMP  */
#define TAIL04_PCX                       21       /* BMP  */
#define TAIL05_PCX                       22       /* BMP  */
#define TAIL06_PCX                       23       /* BMP  */
#define TAIL07_PCX                       24       /* BMP  */
#define TAIL08_PCX                       25       /* BMP  */
#define TAIL09_PCX                       26       /* BMP  */
#define TAIL10_PCX                       27       /* BMP  */
#define TAIL11_PCX                       28       /* BMP  */
#define TAIL12_PCX                       29       /* BMP  */
#define TAIL13_PCX                       30       /* BMP  */
#define TAIL14_PCX                       31       /* BMP  */
#define TAIL15_PCX                       32       /* BMP  */
#define TAIL16_PCX                       33       /* BMP  */
#define TAIL17_PCX                       34       /* BMP  */
#define TAIL18_PCX                       35       /* BMP  */
#define TAIL19_PCX                       36       /* BMP  */
#define TAIL20_PCX                       37       /* BMP  */
#define TAIL21_PCX                       38       /* BMP  */
#define TAIL22_PCX                       39       /* BMP  */
#define TAIL23_PCX                       40       /* BMP  */
#define TAIL24_PCX                       41       /* BMP  */
#define TAIL25_PCX                       42       /* BMP  */
#define TILES_PCX                        43       /* BMP  */
#define WINGS00_PCX                      44       /* BMP  */
#define WINGS01_PCX                      45       /* BMP  */
#define WINGS02_PCX                      46       /* BMP  */
#define WINGS03_PCX                      47       /* BMP  */
#define WINGS04_PCX                      48       /* BMP  */
#define WINGS05_PCX                      49       /* BMP  */
#define WINGS06_PCX                      50       /* BMP  */
#define WINGS07_PCX                      51       /* BMP  */
#define WINGS08_PCX                      52       /* BMP  */
#define WINGS09_PCX                      53       /* BMP  */
#define WINGS10_PCX                      54       /* BMP  */
#define WINGS11_PCX                      55       /* BMP  */

